### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by connor4312**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for creating a TypeScript definition. 🙌 Can you follow [this styleguide](https://github.com/sindresorhus/typescript-definition-style-guide)?

Some examples:
- https://github.com/sindresorhus/semver-regex/commit/039944bb5b3876913171b895488874b5c200e458
- https://github.com/sindresorhus/global-dirs/commit/0aae4d64887f77af94c4fb7ca12aa9f93c0a8291
- https://github.com/sindresorhus/md5-hex/commit/bbbfeff840f26973fbffc16d4863ae6d120079f3

File: index.d.ts [hunk 1]
```diff
@@ -1,3 +1,9 @@
-function lookup(): Promise<{ name: string; id: string }>;
+/**
+ * Gets the default browser on the platform.
+ * @throws if the lookup fails for any reason
```
Comment (sindresorhus) [inline index.d.ts:3]: I don't think this is useful. Pretty much anything could throw in JS and this doesn't really provide any useful info over that.

File: index.d.ts [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
+ * @returns a promise containing the human-readable name and bundle ID of
+ * the discovered browser.
+ */
+declare function lookup(): Promise<{ name: string; id: string }>;
```
Comment (sindresorhus) [inline index.d.ts:7]: I would make a separate type for the return object.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+export = lookup;
```
Comment (sindresorhus) [inline index.d.ts:9]:
```suggestion
export = defaultBrowser;
```

File: package.json [hunk 4]
```diff
@@ -41,8 +42,10 @@
   },
   "devDependencies": {
     "ava": "*",
+    "tsd": "^0.11.0",
     "xo": "*"
   },
+  "types": "./index.d.ts",
```
Comment (sindresorhus) [inline package.json:48]: This isn't needed. See the styleguide.

**Comment by sindresorhus** [comment]
Please look at the styleguide a little bit more thoroughly.

**Comment by sindresorhus** [comment]
@connor4312 Friendly bump :)

**Comment by connor4312** [comment]
Been a busy month in VS Code land. The first week of every month-long iteration is debt week; I will take care of this in the upcoming debt week in August 🙂

**Comment by connor4312** [comment]
I've updated the PR. However I'm having a difficult time trying to find the exact set of ava/xo versions needed to make the build work while keeping the Node 4 target. Since https://github.com/sindresorhus/default-browser/pull/6 updates to Node 8, I left things as they were.

**Pull request merged**